src/helpers/qa.ts

```diff
@@ -0,0 +1,659 @@
+import {
+  Chain,
+  ChainId,
+  ChainType,
+  HashflowApi,
+  validateChain,
+} from '@hashflow/taker-js';
+import {
+  validateEvmAddress,
+  validateSolanaAddress,
+} from '@hashflow/taker-js/dist/helpers/validation';
+import BigNumber from 'bignumber.js';
+import { computeLevelsQuote } from 'helpers/levels';
+import { convertFromDecimals, convertToDecimals } from 'helpers/token';
+import { Environment, PriceLevel, Token } from 'helpers/types';
+import { validateEnvironment, validateMakerName } from 'helpers/validation';
+
+async function fetchQuoteChains(
+  hashflow: HashflowApi,
+  chain: Chain
+): Promise<Chain[]> {
+  process.stdout.write(`Fetching all available quote chains...`);
+  const tradingPairs = await hashflow.getTradingPairs(chain);
+  const chains = tradingPairs.pairs.map(p => p.quoteToken.chain);
+  const seen: string[] = [];
+  const dedupChains: Chain[] = [];
+  chains.forEach(chain => {
+    if (!seen.includes(stringifyChain(chain))) {
+      dedupChains.push(chain);
+      seen.push(stringifyChain(chain));
+    }
+  });
+  return dedupChains;
+}
+
+function stringifyChain(chain: Chain): string {
+  return `${chain.chainType}_${chain.chainId}`;
+}
+
+type QaOptions = {
+  maker: string;
+  chain: number;
+  chain_type: string;
+  quote_chain?: number;
+  quote_chain_type: string;
+  check_all_xchain: boolean;
+  base_token?: string;
+  quote_token?: string;
+  env: string;
+  num_requests: number;
+  delay_ms: number;
+};
+
+export async function runQa(
+  argv: QaOptions,
+  name: string,
+  authKey: string
```

Comment:
Let's make this a proper API call and pass in actual typed arguments rather than just `argv` / `QaOptions` where everything is just `string` or `number`.

Reply:
fixed